### DISCUSSION TITLE
The software is very beautiful, but can some functions be made into plugins?

### DISCUSSION THREAD

**Discussion by lvtao-net** [zed-industries/zed#12364] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

Channels and chats, as a developer, I think the lightweight nature of the software is more suitable, such features can be made into plugins.

**Comment by lvtao-net** [comment]
As a user in the Chinese mainland, OpenAI cannot be used normally. It would be best to be able to configure the API domain. Of course, it would be even better if there were other AI plugins available. For example, Groq or Gemini.

**Comment by jansol** [comment]
It is already possible to change the API endpoint, see https://zed.dev/docs/assistant-panel#using-a-custom-api-endpoint-for-openai

**Comment by Moshyfawn** [comment]
One of the pillars of the Zed concept is its "multiplayer" nature. I doubt that shipping channels as a plugin would be something the Zed team would consider.

<img width="502" alt="image" src="https://github.com/zed-industries/zed/assets/16290753/150a8376-289a-454c-9f26-60eed22beb43">

**Comment by Moshyfawn** [comment]
Related:
- #7121

**Comment by iamnbutler** [comment]
I can say with near absolute certainty that collaboration in zed will never be made into an extension or be disabled by default, as the core of the company is build on the foundation of collaborative editing.

That being said, the project is OSS, I'm sure someone has a zed-no-collab fork going somewhere :)

Closing, but feel free to discuss further in a discussion.

**Comment by pdenapo** [comment]
I would like to be able to build a light-weighted version of Zed without any collaborative features or IA (assistant, copilot). For my use case, the IA features are of no use (I don't have any Open IA/github copilot key), and I mostly program alone. But I do care about memory footprint (I usually need to ran an editor in systems with limited resources or open many instances of the editor).
All I would love to have is a powerful fast and reliable editor for programmers, if possible with IDE capacities (like debugging). I guess that many other users might be in a similar situation. 

So I would love if this features that are consuming my RAM for something that I don't need could be made optional.
Also, if this features are moved to extensions, that would mean fewer dependencies for the core of Zed and fastest compilation times (which impact in the modify-compile-test cycle, hence helping to a more rapid development of Zed itself)
